packages/react-components/lib/tasks/task-timeline.tsx

```diff
@@ -48,47 +47,99 @@ const StyledTimeLine = styled((props: TimeLinePropsWithRef) => <Timeline {...pro
   }),
 );
 
+function nestedEvents(eventsStates: { [key: string]: EventState } | undefined, child: number) {
+  if (eventsStates !== undefined) {
+    const deps = eventsStates[child] ? eventsStates[child].deps : [];
+    return (
+      <TreeItem
+        key={`event-${child}`}
+        nodeId={`parent-${eventsStates[child].name}`}
+        label={eventsStates[child].name}
+      >
+        {deps?.map((id) => {
+          if (eventsStates[id].deps?.length) {
+            nestedEvents(eventsStates, id);
+          } else {
+            return (
+              <TreeItem
+                key={eventsStates[child].name}
+                nodeId={`child-${id}`}
+                label={eventsStates[id].name}
+              ></TreeItem>
+            );
+          }
+        })}
+      </TreeItem>
+    );
+  }
+}
+
+function colorDot(eventsStates: { [key: string]: EventState } | undefined) {
+  let standby: Boolean = true;
+  for (const id in eventsStates) {
+    const event: EventState = eventsStates[id];
+    if (event.status === 'underway') {
+      return 'success' as const;
```

Comment:
I don't quite understand this cast, it seems redundant as `'success'` is already a valid return type (assuming the return is meant to be a string)

Reply:
It used to not accept the string type when it was part of the `dashboard package`, but now after moving it to react components and building it, it works without casting, no idea why. but I just removed the casting.